### PR TITLE
V4 fixes for some plugins and linter warning fixes

### PR DESCRIPTION
**Pull request by buschmann23**

### CODE REVIEW

**Comment by buschmann23** [comment]
Done. Never thought about it. :)

File: Cutelyst/Plugins/CSRFProtection/csrfprotection.h [hunk 1]
```diff
@@ -369,9 +371,10 @@ class CUTELYST_PLUGIN_CSRFPROTECTION_EXPORT CSRFProtection : public Plugin
     static bool checkPassed(Context *c);
 
 protected:
-    CSRFProtectionPrivate *d_ptr;
+    bool setup(Application *app) override;
 
-    virtual bool setup(Application *app) override;
+private:
+    const QScopedPointer<CSRFProtectionPrivate> d_ptr;
```
Comment (dantti) [inline Cutelyst/Plugins/CSRFProtection/csrfprotection.h:377]: better use std::unique_ptr no?

Comment (buschmann23) [reply Cutelyst/Plugins/CSRFProtection/csrfprotection.h:377]: Yes, will change it.

**Comment by dantti** [comment]
LGTM but has conflicts, maybe due merge commits

**Comment by dantti** [comment]
Rebase instead of merging

**Pull request merged**